src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -383,6 +394,35 @@ private void operationAction(TableReference tableRef) throws InterruptedExceptio
       queryJob.waitFor();
     }
 
+    private void updateTableSchema(TableReference tableRef) {
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      TableId sourceTableId = TableId.of(temporaryTableReference.getDatasetId(), temporaryTableReference.getTableId());
+      TableId destinationTableId = TableId.of(tableRef.getDatasetId(), tableRef.getTableId());
+
+      com.google.cloud.bigquery.Table sourceTable = bigquery.getTable(sourceTableId);
+      com.google.cloud.bigquery.Table destinationTable = bigquery.getTable(destinationTableId);
+
+      FieldList sourceFields = sourceTable.getDefinition().getSchema().getFields();
+      tableFieldsList = sourceFields.stream().map(Field::getName).collect(Collectors.toList());
+      FieldList destinationFields = destinationTable.getDefinition().getSchema().getFields();
+
+      List<String> fieldsName = destinationFields.stream().map(Field::getName).collect(Collectors.toList());
+
+      List<Field> newFields = sourceFields.stream().filter(field -> !fieldsName.contains(field.getName()))
+        .collect(Collectors.toList());
+
+      if (!newFields.isEmpty()) {
```

Comment:
update also allows a non-nullable field to be made nullable, which will not satisfy this condition.

Reply:
Added possibility change a non-nullable field to a nullable one.